src/Entity/Member.php

```diff
@@ -0,0 +1,165 @@
+    collectDenormalizationErrors: true
+)]
+#[ApiFilter(filterClass: SearchFilter::class, properties: ['user', 'event', 'payer'])]
+#[ORM\Entity(repositoryClass: MemberRepository::class)]
+#[ORM\UniqueConstraint(columns: ['user_uid', 'event_uid'])]
+class Member
+{
+    #[ORM\Id]
+    #[ORM\Column(length: 255, unique: true)]
+    #[Assert\NotBlank]
+    #[Groups(['member:create', 'member:read'])]
+    private string $uid;
+
+    #[ORM\ManyToOne]
+    #[ORM\JoinColumn(name: 'user_uid', referencedColumnName: 'uid', nullable: false, onDelete: "CASCADE")]
+    #[Assert\NotBlank]
+    #[Groups(['member:create', 'member:read'])]
+    private User $user;
+
+    #[ORM\ManyToOne]
+    #[ORM\JoinColumn(name: 'event_uid', referencedColumnName: 'uid', nullable: false, onDelete: "CASCADE")]
+    #[Assert\NotBlank]
+    #[Groups(['member:create', 'member:read'])]
+    private PartyEvent $event;
+
+    #[ORM\ManyToOne(targetEntity: Member::class, inversedBy: 'payedFor')]
+    #[ORM\JoinColumn(name: 'payer_uid', referencedColumnName: 'uid', onDelete: "set null")]
+    #[EventMember]
+    private ?Member $payer = null;
+
+    #[ORM\OneToMany(mappedBy: 'payer', targetEntity: Member::class, fetch: 'EAGER')]
+    #[Groups(['member:create', 'member:update', 'member:read'])]
+    #[MemberIsNotPayer]
+    #[MemberAndPayerSameEvent]
+    private Collection $payedFor;
+
+    public function __construct($uid)
+    {
+        $this->uid = $uid;
+        $this->payedFor = new ArrayCollection();
+    }
+
+    public function getUid(): string
+    {
+        return $this->uid;
+    }
+
+    public function getUser(): User
+    {
+        return $this->user;
+    }
+
+    public function setUser(User $user): static
+    {
+        $this->user = $user;
+
+        return $this;
+    }
+
+    public function getEvent(): PartyEvent
+    {
+        return $this->event;
+    }
+
+    public function setEvent(PartyEvent $event): static
+    {
+        $this->event = $event;
+
+        return $this;
+    }
+
+    /**
+     * @return Member|null
+     */
+    public function getPayer(): ?Member
+    {
+        return $this->payer;
+    }
+
+    /**
+     * @param Member|null $payer
+     */
+    public function setPayer(?Member $payer): void
+    {
+        $this->payer = $payer;
+    }
+
+    /**
+     * @param array $payedFor
+     * @return Member
+     */
+    public function setPayedFor(array $payedFor): Member
+    {
+        $payedForUids = [];
+        /** @var Member $member */
+        foreach ($payedFor as $member) {
+            $payedForUids[] = $member->getUid();
+            if (!$this->payedFor->contains($member)) {
+                $this->payedFor->add($member);
+                $member->setPayer($this);
+            }
+        }
+        foreach ($this->payedFor as $member) {
+            if (!in_array($member->getUid(), $payedForUids)
+                && $this->payedFor->removeElement($member)
+                && $member->getPayer() === $this
+            ) {
```

Comment:
### ☝ Suggestion: Split this if

I believe u don't need to put all operations and conditions in one array... 
```suggestion
   $payerIsDeleted = !in_array($member->getUid(), $payedForUids);
   if($payerIsDeleted) continue;
   
   $isMemberRemoved = $this->payedFor->removeElement($member);
   if(!$isMemberRemoved) throw new Error(); // not shure we need this check
   
   $memberHasSamePayer = $member->getPayer() === $this;
   if($memberHasSamePayer) $member->setPayer(null);
```